modules/light-clients/08-wasm/types/querier_test.go

```diff
@@ -107,3 +113,88 @@ func (suite *TypesTestSuite) TestCustomQuery() {
 		})
 	}
 }
+
+func (suite *TypesTestSuite) TestStargateQuery() {
+	typeURL := "/ibc.lightclients.wasm.v1.Query/Checksums"
+
+	testCases := []struct {
+		name     string
+		malleate func()
+	}{
+		{
+			"success: custom query",
+			func() {
+				querierPlugin := types.QueryPlugins{
+					Stargate: types.AcceptListStargateQuerier([]string{typeURL}),
+				}
+
+				types.SetQueryPlugins(&querierPlugin)
+
+				suite.mockVM.RegisterQueryCallback(types.StatusMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, querier wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					queryRequest := types.QueryChecksumsRequest{}
+					bz, err := queryRequest.Marshal()
+					suite.Require().NoError(err)
+
+					resp, err := querier.Query(wasmvmtypes.QueryRequest{
+						Stargate: &wasmvmtypes.StargateQuery{
+							Path: typeURL,
+							Data: bz,
+						},
+					}, math.MaxUint64)
+					suite.Require().NoError(err)
+
+					var respData types.QueryChecksumsResponse
+					err = respData.Unmarshal(resp)
+					suite.Require().NoError(err)
+
+					expChecksum := hex.EncodeToString(suite.checksum)
+
+					suite.Require().Len(respData.Checksums, 1)
+					suite.Require().Equal(expChecksum, respData.Checksums[0])
+
+					return resp, wasmtesting.DefaultGasUsed, nil
+				})
+			},
+		},
+		{
+			"failure: default querier",
+			func() {
+				suite.mockVM.RegisterQueryCallback(types.StatusMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, querier wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					queryRequest := types.QueryChecksumsRequest{}
+					bz, err := queryRequest.Marshal()
+					suite.Require().NoError(err)
+
+					resp, err := querier.Query(wasmvmtypes.QueryRequest{
+						Stargate: &wasmvmtypes.StargateQuery{
+							Path: typeURL,
+							Data: bz,
+						},
+					}, math.MaxUint64)
+					suite.Require().ErrorIs(err, wasmvmtypes.UnsupportedRequest{Kind: fmt.Sprintf("'%s' path is not allowed from the contract", typeURL)})
+					suite.Require().Nil(resp)
+
+					return nil, wasmtesting.DefaultGasUsed, err
+				})
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupWasmWithMockVM()
+
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
+
+			tc.malleate()
+
+			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
+			clientState := endpoint.GetClientState()
+			clientState.Status(suite.chainA.GetContext(), clientStore, suite.chainA.App.AppCodec())
+
+			// reset query plugins after each test
+			types.SetQueryPlugins(types.NewDefaultQueryPlugins())
```

Comment:
Why not do this after the call to `SetupWasmWithMockVM` as you do in the `TestCustomQuery`?

Reply:
I think the end of test might be more reliable now. I should maybe change the other one too